daemon/api.go

```diff
@@ -687,6 +688,17 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
 	case store.ErrUnauthenticated, store.ErrInvalidCredentials:
 		return Unauthorized(err.Error())
 	default:
+		if e, ok := err.(*url.Error); ok {
+			if neterr, ok := e.Err.(*net.OpError); ok {
```

Comment:
Do we need the same if we want to handle `snap install` gracefully?

Reply:
No, afacit from quickly looking, install is different because on the api side we only call `snapstate.Install` and get a taskset (change), and it's a `download` task that will fail. In general we have many different error paths in the code and they all need to be investigated and addressed case-by-case if we want this to be robust everywhere :(